### PR TITLE
Branch UI total cost

### LINKED ISSUE #163: Implement "Calculate total" functionality
Users should be able to calculate cash flow (within time range)

### PR DESCRIPTION
**Pull request by quannam0124**
Add Total Transaction box in UI to view the net amount transacted with all clients in the current displayed list.
Implemented calculateTotalTransaction() method in Logic to calculate net amount transacted with all clients in displayed list

### CODE REVIEW

**Review by ElginL — APPROVED** [review]
Tested and looked through it, LGTM!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -147,4 +148,18 @@ public boolean equals(Object obj) {
                 && filteredClientsList.equals(other.filteredClientsList);
     }
 
+    /**
+     * Calculate the total transactions made with clients in the given filtered Client list.
+     *
+     * @param filteredClientList list of Clients
```
Comment (Jo3LW) [inline src/main/java/seedu/address/model/ModelManager.java]: Great javadocs! Maybe making it consistent for clients whether it should be capitalised would be great :) See lines 152 and 154.

**Review by Jo3LW — APPROVED** [review]
LGTM! Well done.

File: src/main/java/seedu/address/logic/Logic.java [hunk 2]
```diff
@@ -47,4 +47,9 @@ public interface Logic {
      * Set the user prefs' GUI settings.
      */
     void setGuiSettings(GuiSettings guiSettings);
+
+    /**
+     * Returns the net amount transacted by all clients in the current list
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/Logic.java]: Perhaps there should be a fullstop after this sentence.

File: src/main/java/seedu/address/model/client/UniqueClientList.java [hunk 3]
```diff
@@ -135,4 +135,18 @@ private boolean clientsAreUnique(List<Client> clients) {
         }
         return true;
     }
+
+    /**
+     * Returns the net amount transacted with all clients shown in the current list
+     * @return
```
Comment (notle1706) [inline src/main/java/seedu/address/model/client/UniqueClientList.java]: Refer to previous comment and I think you might be missing the return statement.

File: src/main/java/seedu/address/ui/NetTransactionBox.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    private static final String FXML = "NetTransactionBox.fxml";
+    private double netTransactionAmount;
+    @FXML
+    private TextField netTransaction;
+
+    /**
+     * Creates a new NetTransactionBox
+     *
+     * @param amount to be stored as the calculated total amount
```
Comment (notle1706) [inline src/main/java/seedu/address/ui/NetTransactionBox.java]: Refer to previous comment.

**Review by notle1706 — CHANGES_REQUESTED** [review]
Some javadocs should be corrected otherwise, good job!

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (quannam0124) [reply src/main/java/seedu/address/model/ModelManager.java]: Thanks for highlighting this!

File: src/main/java/seedu/address/logic/Logic.java (hunk 2 above)
Comment (quannam0124) [reply src/main/java/seedu/address/logic/Logic.java]: Thank you for pointing this out! I will address the other javadocs comments as well

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 5]
```diff
@@ -78,4 +78,9 @@ public GuiSettings getGuiSettings() {
     public void setGuiSettings(GuiSettings guiSettings) {
         model.setGuiSettings(guiSettings);
     }
+
+    @Override
+    public double calculateTotalTransaction(ObservableList<Client> filteredClientList) {
+        return model.calculateTotalTransaction(filteredClientList);
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/LogicManager.java:84]: Is it possible to add a test to cover this line?

File: src/main/java/seedu/address/model/Model.java [hunk 6]
```diff
@@ -84,4 +84,9 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredClientList(Predicate<Client> predicate);
+
+    /**
+     * Updates the total transaction of the filtered client list
```
Comment (notle1706) [inline src/main/java/seedu/address/model/Model.java]: You might have missed out a fullstop on this javadocs.

**Review by notle1706 — CHANGES_REQUESTED** [review]
Just a few more changes to be made, please check through javadocs to make sure they follow the documentation conventions.

File: src/main/java/seedu/address/model/Model.java (hunk 6 above)
Comment (quannam0124) [reply src/main/java/seedu/address/model/Model.java]: Apologies I have added this as well. Thank you

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 5 above)
Comment (quannam0124) [reply src/main/java/seedu/address/logic/LogicManager.java:84]: I realized my test for this line was private instead of public. It should be covered now

**Review by notle1706 — APPROVED** [review]
LGTM. Goodjob on the changes!

**Pull request merged**